analyzer/src/funTest/assets/projects/external/git-repo-expected-output.yml

```diff
@@ -75,9 +75,7 @@ analyzer:
       homepage_url: ""
       scopes:
       - name: "dependencies"
-        dependencies: []
       - name: "devDependencies"
-        dependencies: []
     - id: "Maven:org.apache.commons:commons-text:1.6"
```

Comment:
Hmm, I'm not sure I like this. I believe it's useful to understand from the output format what the general data structure is. If you remove this, at least in this file it looks like `scopes` is only ever used to list the *names* of the scopes, but not what *dependencies* they contain. I'm curious what @mnonnenmacher thinks about this.

Reply:
BTW, the "understand from the output format what the general data structure is" part also holds true for your other PRs the remove(d) data fields that might be empty.

Reply:
We discussed this a long time ago and the reasoning to keep those fields also if empty was the point Sebastian mentioned, plus that it has value to explicitly say that there are no dependencies (or similar for the other fields, like explicitly state that there have not been any curations applied).
That argument was always based on the idea that users would read those files, or that third-party tools would parse them. Because users might not be familiar with the model or not remember any detail. And third-party tools have to handle that certain properties could be there or not which might make the code more complicated. For example, I don't know if this change would make the PostgreSQL queries I currently write more complex or not, I would have to test this first.
However, I'm not sure how relevant this use case is, but on the other hand I'm also not sure how relevant the reduced file size is. I think that is a change I would have brought up on Slack before working on it.

Reply:
Thanks for explaining the history. Anyhow as there are many other places (even in the same classes) which actually do omit empty property values, I wonder what rule of thumb you guys are following. Obviously it does not seem to apply to all properties, can you please clarify?

Reply:
> I think that is a change I would have brought up on Slack before working on it.

I obviously did not have enough information to make that guess.

Reply:
> like explicitly state that there have not been any curations applied

Just as a reminder, then we need to consider reverting https://github.com/oss-review-toolkit/ort/pull/2807.

Reply:
> Anyhow as there are many other places (even in the same classes)

Hmm, in the `Scope` class you modify no other empty values are omitted, and neither was that the case in the `CuratedPackage` before you changed it.

I believe we mostly only make use of `@JsonInclude(JsonInclude.Include.*)` for issues, or mutually exclusive fields.

Reply:
> Hmm, in the `Scope` class you modify no other empty values are omitted, and neither was that the case in the `CuratedPackage` before you changed it.

..but in `PackageReference`it was...

Reply:
Indeed `PackageReference` is an exception because it appears *very* many times in the transitive dependency tree. And it's just a reference after all, the "real" data is in the de-duplicated list of packages in the end, whose entries should be complete in turn.

Reply:
ok, I'll just close the PRs then. Should I revert my change from yesteray as well?

Reply:
I have a tendency to say "yes". Curations are @mnonnenmacher's area, and he explicitly expressed above that he'd like to keep empty curations.

I'd fine with you, @fviernau, directly pushing the revert commit for f5eb87d626c38460e4e25a5b50ebcff7a1a6315d to `master` without a PR.

Reply:
> I wonder what rule of thumb you guys are following.

There's no rule really, it was a case-by-case decision in the past. I think when fields are omitted from serialization it was either because it made a big difference in file size like in the `PackageReference` example above, or for user facing files like curations, where it's really handy to not have to add all of them.

Like @sschuberth I have a tendency towards "yes", but I don't have a strong opinion on this right now, because the current use of `@JsonInclude` is already a bit inconsistent and does not really follow any rules.

Reply:
For example `AnalyzerResult.issues` should not have this annotation, following the arguments above.

Reply:
Btw, the PostgreSQL queries I mentioned above would not suffer from this change, just tried it.